lib/mixlib/cli.rb

```diff
@@ -153,6 +153,7 @@ def deprecated_option(name,
         long: nil,
         short: nil,
         boolean: false,
+        type: nil,
```

Comment:
Can you add a comment to the block above explaining what `type` is for? Maybe
```
# type<Symbol> :: Validate the option as the specified type. Supports `:numeric` to parse an integer or float.
```

Reply:
Okay, will do thanks!

Reply:
This should really look more like properties in chef/chef, so `type: [ String, nil, true, false ]` type of things, where you just use the `===` operator on an array.  That avoids entirely having to write individual code for `:numeric` and then `:string`, and a mostly infinite whack-a-mole to write a type system.

Reply:
Ultimately the `===` operator in properties comes from here:

https://github.com/chef/chef/blob/master/lib/chef/mixin/params_validate.rb#L435

although I'm not sure that code will help you that much, since it has piles of crazy concerns that would not be valid in this context.

but you can kind of see the shape were it loops across all the values and ultimate test with `===` to see if the property is valid against the `is:` list which is passed (`property :foo, [ String, true, false ]` is identical to `property :foo, is: [ String, true, false ]` but the `is:` is conventionally omitted, but the code uses that as the option in the options hash).

Reply:
@lamont-granquist will take a reference from property param validator. Thanks!